UKHO.ExchangeSetService.API/UKHO.ExchangeSetService.Common/Helpers/FileShareService.cs

```diff
@@ -406,14 +426,35 @@ private async Task<bool> ProcessBatchFile(IEnumerable<string> uri, string downlo
+                    BatchId = entry.BatchId,
+                    PartitionKey = productName,
+                    RowKey = $"{editionNumber}|{updateNumber}",
+                    Response = JsonConvert.SerializeObject(entry)
+                };
+                logger.LogInformation(EventIds.FileShareServiceSearchResponseStoreToCacheStart.ToEventId(), "File share service search response insert/merge request in azure table for cache started for Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with FSS BatchId:{FssBatchId}. BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", productName, editionNumber, updateNumber, entry.BatchId, queueMessage.BatchId, queueMessage.CorrelationId);
+                await fileShareServiceCache.InsertOrMergeFssCacheDetail(fssSearchResponseCache);
+                logger.LogInformation(EventIds.FileShareServiceSearchResponseStoreToCacheCompleted.ToEventId(), "File share service search response insert/merge request in azure table for cache completed for Product/CellName:{ProductName}, EditionNumber:{EditionNumber} and UpdateNumber:{UpdateNumber} with FSS BatchId:{FssBatchId}. BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", productName, editionNumber, updateNumber, entry.BatchId, queueMessage.BatchId, queueMessage.CorrelationId);
+            }
             return result;
         }
 
-        private async Task CopyFileToFolder(HttpResponseMessage httpResponse, string path)
+        private async Task CopyFileToFolder(HttpResponseMessage httpResponse, string path, string fileName, BatchDetail entry, SalesCatalogueServiceResponseQueueMessage queueMessage)
         {
-            using (Stream stream = await httpResponse.Content.ReadAsStreamAsync())
+            byte[] bytes = fileSystemHelper.ConvertStreamToByteArray(await httpResponse.Content.ReadAsStreamAsync());
+            fileSystemHelper.CreateFileCopy(path, new MemoryStream(bytes));
+            if (!entry.IgnoreCache && fssCacheConfiguration.Value.IsFssCacheEnabled)
             {
-                fileSystemHelper.CreateFileCopy(path, stream);
+                logger.LogInformation(EventIds.FileShareServiceUploadENCFilesToCacheStart.ToEventId(), "File share service upload ENC file request to cache blob container started for Container:{Container}, with FileName: {FileName}. ESS BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", entry.BatchId, fileName, queueMessage.BatchId, queueMessage.CorrelationId);
+                await fileShareServiceCache.CopyFileToBlob(new MemoryStream(bytes), fileName, entry.BatchId);
+                logger.LogInformation(EventIds.FileShareServiceUploadENCFilesToCacheCompleted.ToEventId(), "File share service upload ENC file request to cache blob container completed for Container:{Container}, with FileName: {FileName}. ESS BatchId:{batchId} and _X-Correlation-ID:{CorrelationId}", entry.BatchId, fileName, queueMessage.BatchId, queueMessage.CorrelationId);
```

Comment:
There is no test coverage for this new code.

Reply:
We will do this test code coverage in a separate Miscellaneous Task on Monday